### PR TITLE
Ensure status code is correctly set for default response

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/NSwag.SwaggerGeneration.AspNetCore/Processors/OperationResponseProcessor.cs [hunk 1]
```diff
@@ -75,10 +75,11 @@ public OperationResponseProcessor(AspNetCoreToSwaggerGeneratorSettings settings)
                     var returnType = apiResponse.Type;
                     var response = new SwaggerResponse();
                     string httpStatusCode;
-                    if (apiResponse.StatusCode == 0 && IsVoidResponse(returnType))
-                        httpStatusCode = "200";
-                    else if (apiResponse.TryGetPropertyValue<bool>("IsDefaultResponse"))
+
+                    if (apiResponse.TryGetPropertyValue<bool>("IsDefaultResponse"))
```
Comment (pranavkm) [inline src/NSwag.SwaggerGeneration.AspNetCore/Processors/OperationResponseProcessor.cs]: `IsDefaultResponse` needs to be prioritized over the "0" status code. Added a couple of unit tests to cover the behavior of `ProcessAsync`

Comment (RSuter) [reply src/NSwag.SwaggerGeneration.AspNetCore/Processors/OperationResponseProcessor.cs]: Thanks for the PR. Can you delete UpgradeLog2.htm? Maybe you should work in NSwag.Min.sln, this one is less heavy...

Comment (RSuter) [reply src/NSwag.SwaggerGeneration.AspNetCore/Processors/OperationResponseProcessor.cs]: Everything looks good, thank you!

Comment (pranavkm) [reply src/NSwag.SwaggerGeneration.AspNetCore/Processors/OperationResponseProcessor.cs]: Done! I think it did not have the AspNetCore.Tests project so I had to open the other one.

**Pull request merged**